### PR TITLE
Arc 252 carma j2735

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
This PR is to include a prototype implementation of using carma-j2735 library instead of manually generated asn1c code for the J2735 2026 r63 version of the standard

## Related Issue
[ARC-252](https://usdot-carma.atlassian.net/browse/ARC-252)

## Motivation and Context
Use consolidated library for UPER encoding/decoding

## How Has This Been Tested?
Unit testing 

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[ARC-252]: https://usdot-carma.atlassian.net/browse/ARC-252?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/tmx/CMakeLists.txt [hunk 1]
```diff
@@ -43,8 +47,9 @@ LINK_DIRECTORIES (${PROJECT_NAME} "${TMX_LIB_DIR}")
 #
 # Add the generated SAE J2735 library and an interface that contains the build properties
 #
-ADD_SUBDIRECTORY (Asn_J2735)
-ADD_LIBRARY (libasn_j2735 ALIAS ${ASN_J2735_LIBRARIES})
+# ADD_SUBDIRECTORY (Asn_J2735)
```
Comment (paulbourelly999) [inline src/tmx/CMakeLists.txt:50]: We can probably remove this line completely

File: src/tmx/CMakeLists.txt [hunk 2]
```diff
@@ -43,8 +47,9 @@ LINK_DIRECTORIES (${PROJECT_NAME} "${TMX_LIB_DIR}")
...
+ADD_LIBRARY (libasn_j2735 ALIAS ::carma_j2735)
```
Comment (paulbourelly999) [inline src/tmx/CMakeLists.txt:51]: We also do not need an alias for this library

File: src/tmx/TmxApi/CMakeLists.txt [hunk 3]
```diff
@@ -43,8 +43,11 @@ IF (NOT TMXAPI_NO_STATIC)
     ENDIF ()
     TARGET_INCLUDE_DIRECTORIES (${PROJECT_NAME}Static SYSTEM PUBLIC
                                 $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+                                /opt/carma/include/carma_j2735
                                 ${Boost_INCLUDE_DIRS}) 
-    TARGET_LINK_LIBRARIES (${PROJECT_NAME}Static ${ASN_J2735_LIBRARIES} ${Boost_LIBRARIES} ${ICU_LIBRARIES} pugixml)
+    TARGET_LINK_LIBRARIES (${PROJECT_NAME}Static libasn_j2735 ${Boost_LIBRARIES} ${ICU_LIBRARIES} pugixml)
```
Comment (paulbourelly999) [inline src/tmx/TmxApi/CMakeLists.txt:48]: lets link ::carma_j2735 directly instead of the alias

File: src/tmx/TmxApi/CMakeLists.txt [hunk 4]
```diff
@@ -63,12 +66,15 @@ ENDIF (NOT TMXAPI_NO_STATIC)
...
     IF (TMX_LIB_DIR)
-        SET_TARGET_PROPERTIES (${PROJECT_NAME} PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${TMX_LIB_DIR}")
+        SET_TARGET_PROPERTIES (${PROJECT_NAME} PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${TMX_LIB_DIR}" INSTALL_RPATH "/opt/carma/lib")
     ENDIF ()
     TARGET_INCLUDE_DIRECTORIES (${PROJECT_NAME} SYSTEM PUBLIC 
                                 $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+                                /opt/carma/include/carma_j2735
                                 ${Boost_INCLUDE_DIRS})
-    TARGET_LINK_LIBRARIES (${PROJECT_NAME} PUBLIC ${ASN_J2735_LIBRARIES} ${Boost_LIBRARIES} ${ICU_LIBRARIES} pugixml)
+    TARGET_LINK_LIBRARIES (${PROJECT_NAME} PUBLIC libasn_j2735 ${Boost_LIBRARIES} ${ICU_LIBRARIES} pugixml)
```
Comment (paulbourelly999) [inline src/tmx/TmxApi/CMakeLists.txt:75]: Lets link ::carma_j2735 instead of the alias

File: src/tmx/TmxCore/CMakeLists.txt [hunk 5]
```diff
@@ -12,7 +12,9 @@ TARGET_INCLUDE_DIRECTORIES( ${PROJECT_NAME} PUBLIC
                             ${MYSQL_INCLUDE_DIRS} ${MYSQLCPPCONN_INCLUDE_DIRS} )
 TARGET_LINK_LIBRARIES ( ${PROJECT_NAME} PUBLIC ${TMXAPI_LIBRARIES} )
 TARGET_LINK_LIBRARIES ( ${PROJECT_NAME} PUBLIC ${MYSQL_LIBRARIES} ${MYSQLCPPCONN_LIBRARIES} )
-TARGET_LINK_LIBRARIES ( ${PROJECT_NAME} PUBLIC pthread m rt )
+TARGET_LINK_LIBRARIES ( ${PROJECT_NAME} PUBLIC libasn_j2735 pthread m rt )
```
Comment (paulbourelly999) [inline src/tmx/TmxCore/CMakeLists.txt:15]: Lets link ::carma_j2735 directly instead of the alias

File: src/tmx/CMakeLists.txt (hunk 1 above)
Comment (dan-du-car) [reply src/tmx/CMakeLists.txt]: Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/654/commits/ef1238953d99aa1f7aa13b59a7133c1131039ee5

File: src/tmx/CMakeLists.txt (hunk 2 above)
Comment (dan-du-car) [reply src/tmx/CMakeLists.txt]: Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/654/commits/ef1238953d99aa1f7aa13b59a7133c1131039ee5

File: src/tmx/TmxApi/CMakeLists.txt (hunk 3 above)
Comment (dan-du-car) [reply src/tmx/TmxApi/CMakeLists.txt]: Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/654/commits/ef1238953d99aa1f7aa13b59a7133c1131039ee5

File: src/tmx/TmxApi/CMakeLists.txt (hunk 4 above)
Comment (dan-du-car) [reply src/tmx/TmxApi/CMakeLists.txt]: Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/654/commits/ef1238953d99aa1f7aa13b59a7133c1131039ee5

File: src/tmx/TmxCore/CMakeLists.txt (hunk 5 above)
Comment (dan-du-car) [reply src/tmx/TmxCore/CMakeLists.txt:15]: Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/654/commits/ef1238953d99aa1f7aa13b59a7133c1131039ee5

**Review by radishco — APPROVED** [review]

**Review by dan-du-car — APPROVED** [review]

**Pull request merged**